test/tbb/test_parallel_phase.cpp

```diff
@@ -0,0 +1,273 @@
+#define TBB_PREVIEW_PARALLEL_PHASE 1
+
+#include "common/test.h"
+#include "common/utils.h"
+#include "common/spin_barrier.h"
+
+#include "tbb/task_arena.h"
+#include "tbb/parallel_for.h"
+
+struct dummy_func {
+    void operator()() const {
+    }
+};
+
+template <typename F1 = dummy_func, typename F2 = dummy_func>
+std::size_t measure_median_start_time(tbb::task_arena* ta, const F1& start = F1{}, const F2& end = F2{}) {
+    std::size_t num_threads = ta ? ta->max_concurrency() : tbb::this_task_arena::max_concurrency();
+    std::size_t num_runs = 1000;
+    std::vector<std::size_t> longest_start_times;
+    longest_start_times.reserve(num_runs);
+
+    std::vector<std::chrono::steady_clock::time_point> start_times(num_threads);
+    utils::SpinBarrier barrier(num_threads);
+    auto measure_start_time = [&] (std::size_t) {
+        start_times[tbb::this_task_arena::current_thread_index()] = std::chrono::steady_clock::now();
+        barrier.wait();
+    };
+
+    auto get_longest_start = [&] (std::chrono::steady_clock::time_point start_time) {
+        std::size_t longest_time = 0;
+        for (auto& time : start_times) {
+            auto diff = std::chrono::duration_cast<std::chrono::microseconds>(time - start_time);
+            longest_time = std::max(longest_time, std::size_t(diff.count()));
+        }
+        return longest_time;
+    };
+
+    auto work = [&] {
+        auto start_time = std::chrono::steady_clock::now();
+        start();
+        tbb::parallel_for(std::size_t(0), num_threads, measure_start_time, tbb::static_partitioner{});
+        end();
+        longest_start_times.push_back(get_longest_start(start_time));
+    };
+
+    for (std::size_t i = 0; i < num_runs; ++i) {
+        if (ta) {
+            ta->execute(work);
+        } else {
+            work();
+        }
+        utils::doDummyWork(i*500);
+    }
+    return utils::median(longest_start_times.begin(), longest_start_times.end());
```

Comment:
Doesn't "**_i_** * 500" mean that earlier runs will take less to restart than later runs? Would it make more sense to keep the amount of dummy work the same across each run?

If this increasing non-TBB work amount is intentional, wouldn't it eventually become larger than the current block time of 1ms?

Reply:
@dnmokhov Changed the constant.